backend/data/processors/program_conditions_tokenising.py

```diff
@@ -0,0 +1,162 @@
+"""
+Documentation: TBA
+
+Assume everything is a maturity condition for now.
+- Tokenise the pre-processed data.
+"""
+
+from contextlib import suppress
+import re
+from typing import Dict, Iterator, List, Optional
+from data.processors.program_conditions_pre_processing import PRE_PROCESSED_DATA_PATH
+from data.utility.data_helpers import read_data, write_data
+
+FINAL_TOKENS_PATH = "data/final_data/programsConditionsTokens.json"
+
+
+def tokenise_program_conditions():
+    """
+    """
+    pre_processed_conditions: Dict[str, List[Dict[str, str]]] = read_data(PRE_PROCESSED_DATA_PATH)
+
+    tokenised_conditions: Dict[str, List[Dict[str, str]]] = read_data(PRE_PROCESSED_DATA_PATH)
+
+    # TODO: for loops cringe
+    for program, pre_processed_conditions in pre_processed_conditions.items():
+        cond_list = []
+        for pre_processed_conditions in pre_processed_conditions:
```

Comment:
??????????? are you shadowing `pre_processed_conditons`? Whatever that is is pure cringe

Reply:
Did i count 3 different assignments inside the same for loop?????